# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/src/prepare.rs
# Commit: 5e5b664a9b5e
# Full Hash: 5e5b664a9b5ecba443472c38b1100bae3ab55c9e
# Author: Dzmitry Malyshau <dmalyshau@mozilla.com>
# Date: 2020-07-24 09:32:06
# Description:
#   Bug 1654901 - Make WR primitives invisible when failing to prepare for rendering r=gw
#   
#   the problem was that paths that determined visibility were not switching it off
#   on the primitive instance. Now it's moved one level higher.
#   
# ==============================================================================

diff -r ec25c0d069ae -r 5e5b664a9b5e gfx/wr/webrender/src/prepare.rs
--- a/gfx/wr/webrender/src/prepare.rs	Thu Jul 23 20:28:09 2020 +0000
+++ b/gfx/wr/webrender/src/prepare.rs	Thu Jul 23 23:29:54 2020 +0000
@@ -110,6 +110,8 @@
                 tile_caches,
             ) {
                 frame_state.profile_counters.visible_primitives.inc();
+            } else {
+                prim_instance.visibility_info = PrimitiveVisibilityIndex::INVALID;
             }
         }
     }
@@ -163,8 +165,6 @@
                             println!("\tculled for carrying an invisible composite filter");
                         }
 
-                        prim_instance.visibility_info = PrimitiveVisibilityIndex::INVALID;
-
                         return false;
                     }
                 }
